cpp/src/gandiva/precompiled/arithmetic_ops.cc

```diff
@@ -180,4 +180,32 @@ NUMERIC_BOOL_DATE_FUNCTION(IS_NOT_DISTINCT_FROM)
 
 NUMERIC_FUNCTION(DIVIDE)
 
+#define DIV(TYPE)                                               \
+  FORCE_INLINE                                                  \
+  TYPE div_##TYPE##_##TYPE(int64 context, TYPE in1, TYPE in2) { \
+    if (in2 == 0) {                                             \
+      char const* err_msg = "divide by zero error";             \
+      gdv_fn_context_set_error_msg(context, err_msg);           \
+      return 0;                                                 \
+    }                                                           \
+    return static_cast<TYPE>(in1 / in2);                        \
+  }
+
+DIV(int32)
+DIV(int64)
+
+#define DIV_FLOAT(TYPE)                                         \
+  FORCE_INLINE                                                  \
+  TYPE div_##TYPE##_##TYPE(int64 context, TYPE in1, TYPE in2) { \
+    if (in2 == 0) {                                             \
+      char const* err_msg = "divide by zero error";             \
+      gdv_fn_context_set_error_msg(context, err_msg);           \
+      return 0;                                                 \
+    }                                                           \
+    return static_cast<TYPE>(trunc(in1 / in2));                 \
```

Comment:
please qualify with the namespace for trunc..

Reply:
done